Add admin page

gellend opened the pull request:
Changes:
- Create admin login page + api
- Create admin dashboard to see all users registered + api

gellend committed:
Completing login page

I changed some single quotes to double quotes, and followed Ryan's coding style to make it look better

gellend committed:
Create baseUrl in utils

I create baseUrl in utils so we can reuse it later

gellend committed:
Create dummy API for login

Create a dummy API for login using reference from Ryan's code

gellend committed:
Create admin login page + api

gellend committed:
Create admin dashboard + api

harryakbar commented on src/pages/admin/index.js:
```diff
@@ -0,0 +1,127 @@
+            </body>
+        </>
+    )
+}
```
minor, add new line

harryakbar commented on src/pages/api/admin.js:
```diff
@@ -0,0 +1,67 @@
+            });
+            break;
+    }
+}
```
add new line

harryakbar commented on src/utils/baseUrl.js:
```diff
@@ -0,0 +1 @@
+export const baseUrl = "http://localhost:3000/api";
```
add new line

harryakbar commented on src/pages/admin/index.js:
```diff
@@ -0,0 +1,127 @@
+                        <Typography component="h1" variant="h5">
+                            Admin Page
+                        </Typography>
+                        <Button color="inherit">Log out</Button>
```
When logout, don't forget redirect to Admin Login page

gellend replied:
I'll update it on the next commit

harryakbar commented on src/pages/admin/index.js:
```diff
@@ -0,0 +1,127 @@
+                                </TableRow>
+                            </TableHead>
+                            <TableBody>
+                                {users.map((user) => (
```
When I tried it, the users is not showed, can you investigate it?

gellend replied:
Seems like the API not running in Netlify mas, I try it in the local, it works perfectly

harryakbar replied:
Yes it because you call hardcoded localhost:3000 when deployed.

How to debug:
1. Open inspect element
2. Click network tab
3. Try when you click login button and watch the api request list on Network tab
4. It will call localhost:3000 which is not found url address

https://user-images.githubusercontent.com/22090926/132224638-4891ff86-6b20-423e-9665-55d16fe1e404.mov

gellend replied:
Should I change it to the domain that Netlify provides for this app?

harryakbar replied:
yes, you can check the discord for another way

harryakbar commented on src/pages/auth/login.js:
```diff
@@ -1,42 +1,85 @@
+            password
+        };
+
+        axios
+            .post(`${baseUrl}/login`, payload)
+            .then((res) => Router.push("/todo-list"))
+            .catch((err) => {
+                setIsOpenAlert(true);
+                console.error(err);
+            });
```
When I tried it, it's fail to login, what is the correct email and password?

harryakbar replied:
should the api url {baseUrl}/adminLogin?

gellend replied:
It's the same problem with the users mapping mas, the API not running

gellend committed:
fix minor update

changes:
- add new line in the end of the file
- update link in the button for redirect
- add some styling in the header

gellend committed:
change value in baseUrl
